ResearchSuite/ResearchSuite/RSDTableItem.swift

```diff
@@ -390,3 +404,162 @@ final class RSDNumberInputTableItem : RSDTextInputTableItem {
+        // Setup the formatter.
+        var formatter: Formatter? = (inputField.range as? RSDRangeWithFormatter)?.formatter
+        var placeholderText: String?
+        if (formatter == nil) {
+            let massFormatter = RSDMassFormatter(forInfantUse: (measurementSize == .infant), unitSymbol: unit)
+            formatter = massFormatter
+        }
+        
+        let massUnit: UnitMass
+        if let massFormatter = formatter as? RSDMassFormatter {
+            // When converting from the value entered by the participant, then the
+            // locale is used to determine the preferred units.
+            massFormatter.unitStyle = .long
+            if Locale.current.usesMetricSystem {
+                placeholderText = massFormatter.unitString(fromValue: 60, unit: .kilogram)
+            } else {
+                placeholderText = massFormatter.unitString(fromValue: 140, unit: .pound)
+            }
+            massFormatter.unitStyle = .medium
+            
+            massUnit = massFormatter.toStringUnit
+        }
+        else {
+            massUnit = ((unit != nil) ? UnitMass(fromSymbol: unit!) : nil) ?? .kilograms
+        }
+        
+        self.baseUnit = massUnit
+        unit = massUnit.symbol
+        
+        // If the measurement is for an infant and the local is US
+        // then use the infant mass picker.
+        var pickerSource: RSDPickerDataSource? = (inputField as? RSDPickerDataSource)
+        if pickerSource == nil, measurementSize == .infant, !Locale.current.usesMetricSystem {
+            pickerSource = RSDUSInfantMassPickerDataSource(formatter: formatter as? RSDMassFormatter)
+        }
+        
+        // Switch the hint type to a supported type most appropriate to the units.
+        var hint: RSDFormUIHint = uiHint
+        if uiHint != .popover {
+            hint = (pickerSource == nil) ? .textfield : .picker
+        }
+
+        let answerType = RSDAnswerResultType(baseType: .decimal, sequenceType: nil, dateFormat: nil, unit: unit, sequenceSeparator: nil)
+        
+        super.init(rowIndex: rowIndex, inputField: inputField, uiHint: hint, answerType: answerType, textFieldOptions: nil, formatter: formatter, pickerSource: pickerSource, placeholderText: placeholderText)
+    }
+    
+    /// Override the `convertAnswer()` function to convert the `Measurement` returned
+    /// by the formatter into a decimal value in the `baseUnit`.
+    override func convertAnswer(_ newValue: Any) throws -> Any? {
+        let answer = try super.convertAnswer(newValue)
+        guard let measurement = answer as? Measurement<UnitMass> else { return answer }
+        return measurement.converted(to: baseUnit).value
+    }
+}
+
+/// A custom implementation for inputing a person's height that can set up a picker for
+/// setting the height of an adult. Typically, US English user's will report an adult height
+/// using ft and in, whereas a child or infant will be reported in inches. The placeholder
```

Comment:
Our doctor records my 54" son's height using inches only. I thought about changing it, but decided that I would implement localization of formatted units instead since a question asked using this type is indicating a range such as, "Enter your child's height" and this needs to be something consistent. In my case, I go look at the report from his last checkup and write down what it says. Forcing the user to convert to 4' 6" is more confusing whereas having a text field input for this case where the text field formatter can convert from 4' 6" is easier for the user.